### PR TITLE
Add Azure lighthouse modules

### PR DESCRIPTION
**Pull request by Fred-sun**
##### SUMMARY

Add new module relate with lighthouse.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

azure_rm_registrationdefinition.py
azure_rm_registrationdefinition_info.py
azure_rm_registrationassignment.py
azure_rm_registrationassignment_info.py
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/azure_rm_registrationdefinition_info.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+    registration_definition_id:
+        description:
+            - Guid of the registration definition.
+        type: str
+extends_documentation_fragment:
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+author:
+    - GuopengLin (@t-glin)
+    - Fred-Sun (@Fred-Sun)
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationdefinition_info.py:37]: Fred is the author for this module

File: plugins/modules/azure_rm_registrationassignment.py [hunk 2]
```diff
@@ -0,0 +1,381 @@
...
+---
+module: azure_rm_registrationassignment
+version_added: '1.3.0'
+short_description: Manage Azure RegistrationAssignment instance
+description:
+    - Create, update and delete instance of Azure RegistrationAssignment.
+options:
+    scope:
+        description:
+            - Scope of the resource.
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py:26]: what is the possible value of the scope option?

File: plugins/modules/azure_rm_registrationassignment.py [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+        required: true
+        type: str
+    registration_assignment_id:
+        description:
+            - Guid of the registration assignment.
+        type: str
+    expand_registration_definition:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py:33]: return the registration definition id is enough

File: plugins/modules/azure_rm_registrationassignment.py [hunk 4]
```diff
@@ -0,0 +1,381 @@
...
+                          returned: always
+                          type: str
+                          sample: null
+                      version:
+                          description:
+                              - The plan's version.
+                          returned: always
+                          type: str
+                          sample: null
+id:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py]: format is not correct, need intent?

File: plugins/modules/azure_rm_registrationassignment.py [hunk 5]
```diff
@@ -0,0 +1,381 @@
...
+    - azure.azcollection.azure
+    - azure.azcollection.azure_tags
+author:
+    - GuopengLin (@t-glin)
+    - Fred-Sun (@Fred-Sun)
+    - Haiyuan Zhang (@haiyuazhang)
+
+'''
+
+EXAMPLES = '''
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py:65]: could we add another example to show that scope can be in the subscription/group level?

File: plugins/modules/azure_rm_registrationassignment.py (hunk 3 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment.py:33]: This is defined in the API, return detail info or not!

File: plugins/modules/azure_rm_registrationassignment.py (hunk 4 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment.py]: Id, properties, type and name are the values in the returned dictionary state. It increases the indentation. Thank you very much!

File: plugins/modules/azure_rm_registrationdefinition_info.py (hunk 1 above)
Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationdefinition_info.py:37]: Ok , I will update!

File: plugins/modules/azure_rm_registrationassignment.py [hunk 6]
```diff
@@ -0,0 +1,381 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2020 GuopengLin, (@t-glin)
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment.py]: should be fred

File: plugins/modules/azure_rm_registrationassignment_info.py [hunk 7]
```diff
@@ -0,0 +1,306 @@
...
+    def __init__(self):
+        self.module_arg_spec = dict(
+            scope=dict(
+                type='str',
+                required=True
+            ),
+            registration_assignment_id=dict(
+                type='str'
+            ),
+            expand_registration_definition=dict(
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment_info.py:121]: make it option, default to false

File: plugins/modules/azure_rm_registrationassignment_info.py [hunk 8]
```diff
@@ -0,0 +1,306 @@
...
+
+    - name: Get Registration Assignments
+      azure_rm_registrationassignment_info:
+        scope: subscription/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+
+
+'''
+
+RETURN = '''
+registration_assignments:
```
Comment (haiyuazhang) [inline plugins/modules/azure_rm_registrationassignment_info.py:61]: check the structure of info module

Comment (Fred-sun) [reply plugins/modules/azure_rm_registrationassignment_info.py:61]: same with other info module!

**Pull request closed without merging**